com.unity.multiplayer.mlapi/Prototyping/NetworkTransform.cs

```diff
@@ -77,70 +184,160 @@ public void NetworkSerialize(NetworkSerializer serializer)
+
+            return
+                // InLocalSpace Check
+                (InLocalSpace != networkState.InLocalSpace) ||
+                // Position Check
+                (SyncPositionX && !Mathf.Approximately(position.x, networkState.PositionX)) ||
+                (SyncPositionY && !Mathf.Approximately(position.y, networkState.PositionY)) ||
+                (SyncPositionZ && !Mathf.Approximately(position.z, networkState.PositionZ)) ||
+                // Rotation Check
+                (SyncRotationX && !Mathf.Approximately(rotation.x, networkState.RotationX)) ||
+                (SyncRotationY && !Mathf.Approximately(rotation.y, networkState.RotationY)) ||
+                (SyncRotationZ && !Mathf.Approximately(rotation.z, networkState.RotationZ)) ||
+                // Scale Check
+                (SyncScaleX && !Mathf.Approximately(scale.x, networkState.ScaleX)) ||
+                (SyncScaleY && !Mathf.Approximately(scale.y, networkState.ScaleY)) ||
+                (SyncScaleZ && !Mathf.Approximately(scale.z, networkState.ScaleZ));
+        }
+
+        internal void UpdateNetworkState()
+        {
+            var position = InLocalSpace ? m_Transform.localPosition : m_Transform.position;
+            var rotation = InLocalSpace ? m_Transform.localRotation : m_Transform.rotation;
+            var scale = InLocalSpace ? m_Transform.localScale : m_Transform.lossyScale;
+
+            // InLocalSpace Bit
+            ReplNetworkState.Value.InLocalSpace = InLocalSpace;
+            // Position Bits
+            (ReplNetworkState.Value.HasPositionX, ReplNetworkState.Value.HasPositionY, ReplNetworkState.Value.HasPositionZ) =
+                (SyncPositionX, SyncPositionY, SyncPositionZ);
+            // Rotation Bits
+            (ReplNetworkState.Value.HasRotationX, ReplNetworkState.Value.HasRotationY, ReplNetworkState.Value.HasRotationZ) =
+                (SyncRotationX, SyncRotationY, SyncRotationZ);
+            // Scale Bits
+            (ReplNetworkState.Value.HasScaleX, ReplNetworkState.Value.HasScaleY, ReplNetworkState.Value.HasScaleZ) =
+                (SyncScaleX, SyncScaleY, SyncScaleZ);
+
+            // Position Values
+            (ReplNetworkState.Value.PositionX, ReplNetworkState.Value.PositionY, ReplNetworkState.Value.PositionZ) =
+                (position.x, position.y, position.z);
+            // Rotation Values
+            (ReplNetworkState.Value.RotationX, ReplNetworkState.Value.RotationY, ReplNetworkState.Value.RotationZ) =
+                (rotation.x, rotation.y, rotation.z);
+            // Scale Values
+            (ReplNetworkState.Value.ScaleX, ReplNetworkState.Value.ScaleY, ReplNetworkState.Value.ScaleZ) =
+                (scale.x, scale.y, scale.z);
+
+            ReplNetworkState.SetDirty(true);
+        }
+
+        // TODO: temporary! the function body below probably needs to be rewritten
+        // (e.g. rotation/quaternion computation looks unreliable, needs to be properly vetted/tested)
```

Comment:
I'd put this onto your plate @SamuelBellomo when you start working on interpolator integration & polish.